### PR TITLE
Comented the code, changed the ndb and created 'usefull links' file

### PR DESCRIPTION
**Pull request by 0Cristofer**
The code commented is here. I made some changes in the EnabledStatus class and in the functions so they looks exactly the way it does in google examples, in core it works the same way, any question contact-me on telegram (@cristoferoswald)

### CODE REVIEW

**Comment by 0Cristofer** [comment]
Another thing, I made "usefull links" file to help too

File: main.py [hunk 1]
```diff
@@ -21,26 +22,32 @@
 
 # ================================
 
-class EnableStatus(ndb.Model):
+class EnableStatus(ndb.Model): #NDB entity called EnabledStatus
```
Comment (yukuku) [inline main.py:25]: typo in comment: should be EnableStatus without 'd'

File: main.py [hunk 2]
```diff
@@ -21,26 +22,32 @@
...
     # key name: str(chat_id)
-    enabled = ndb.BooleanProperty(indexed=False, default=False)
+    enabled = ndb.BooleanProperty(indexed=False, default=False) #Entity has atribute enabled
```
Comment (yukuku) [inline main.py:27]: attribute not atribute

File: main.py [hunk 3]
```diff
@@ -21,26 +22,32 @@
...
 
 
 # ================================
 
 def setEnabled(chat_id, yes):
-    es = EnableStatus.get_or_insert(str(chat_id))
-    es.enabled = yes
+    es = ndb.Key(EnableStatus, str(chat_id)).get() #Gets the entity
```
Comment (yukuku) [inline main.py:33]: do not change this. Please keep it as es = EnableStatus.get_or_insert(str(chat_id))

File: main.py [hunk 4]
```diff
@@ -21,26 +22,32 @@
...
+    if es: #If it exists
+        es.enabled = yes #Sets its enabled atribute
+        es.put()
+        return
+    es = EnableStatus(id = str(chat_id)) #If not creates a new entity
     es.put()
 
 def getEnabled(chat_id):
-    es = EnableStatus.get_by_id(str(chat_id))
+    es = ndb.Key(EnableStatus, str(chat_id)).get()
```
Comment (yukuku) [inline main.py:42]: Do not change this as well

File: main.py [hunk 5]
```diff
@@ -21,26 +22,32 @@
...
     if es:
-        return es.enabled
+        return es.enabled #Return the atual state
```
Comment (yukuku) [inline main.py:44]: actual not atual

File: main.py [hunk 6]
```diff
@@ -21,26 +22,32 @@
...
+    es = EnableStatus(id = str(chat_id))
+    es.put()
     return False
 
 
-# ================================
+# ================================ This part makes the comunication google-telegram
```
Comment (yukuku) [inline main.py:50]: remove the comment please

File: main.py [hunk 7]
```diff
@@ -69,21 +76,22 @@ def post(self):
...
         self.response.write(json.dumps(body))
-
+        #From here you can take message information, now it only uses the chat_id and text,
+        #you can take more things from it, search how to use json on google
         update_id = body['update_id']
         message = body['message']
         message_id = message.get('message_id')
         date = message.get('date')
-        text = message.get('text')
+        text = message.get('text') #Takes the 'text' string
```
Comment (yukuku) [inline main.py:85]: Takes the 'text' property (that is the more proper name in json)

File: main.py [hunk 8]
```diff
@@ -99,40 +107,45 @@ def reply(msg=None, img=None):
...
 
             logging.info('send response:')
             logging.info(resp)
-
+        #From here you can make custom commands, just add an 'elif'
         if text.startswith('/'):
             if text == '/start':
                 reply('Bot enabled')
-                setEnabled(chat_id, True)
+                setEnabled(chat_id, True) #Sets the status to True (read above comments)
```
Comment (yukuku) [inline main.py:119]: Sets the _enabled_ status to True

File: main.py [hunk 9]
```diff
@@ -99,40 +107,45 @@ def reply(msg=None, img=None):
...
             elif text == '/stop':
                 reply('Bot disabled')
-                setEnabled(chat_id, False)
-            elif text == '/image':
-                img = Image.new('RGB', (512, 512))
+                setEnabled(chat_id, False) #Changes it to false
```
Comment (yukuku) [inline main.py:122]: Changes what?

File: main.py [hunk 10]
```diff
@@ -99,40 +107,45 @@ def reply(msg=None, img=None):
...
         elif 'who are you' in text:
             reply('telebot starter kit, created by yukuku: https://github.com/yukuku/telebot')
         elif 'what time' in text:
             reply('look at the top-right corner of your screen!')
         else:
-            if getEnabled(chat_id):
+            if getEnabled(chat_id): #If the status of the bot is enabled the bot answers you
                 try:
-                    resp1 = json.load(urllib2.urlopen('http://www.simsimi.com/requestChat?lc=en&ft=1.0&req=' + urllib.quote_plus(text.encode('utf-8'))))
+                    resp1 = json.load(urllib2.urlopen('http://www.simsimi.com/requestChat?lc=en&ft=1.0&req=' + urllib.quote_plus(text.encode('utf-8')))) #Sends you mesage to simsimi IA
```
Comment (yukuku) [inline main.py:148]: What is IA? I think you should mention it as API.

File: main.py [hunk 11]
```diff
@@ -146,7 +159,7 @@ def reply(msg=None, img=None):
             else:
                 logging.info('not enabled for chat_id {}'.format(chat_id))
 
-
+#Telegram comunication (dont change)
```
Comment (yukuku) [inline main.py:162]: don't not dont

File: usefull links.txt [hunk 12]
```diff
@@ -0,0 +1,5 @@
+This bot is made in python and works on Google app engine server, the best way to understand it is to read Telegram and Google documentation
```
Comment (yukuku) [inline usefull links.txt:1]: Remove this. Don't add new files.

**Comment by yukuku** [comment]
Hi, I have examined your pull request, and put some notes on the code. Please check for typos. Also do not add the new file (usefull links.txt)

**Comment by yukuku** [comment]
Please make another pull request.

File: main.py (hunk 11 above)
Comment (ADTC) [inline main.py:162]: communication not comunication

File: main.py (hunk 9 above)
Comment (ADTC) [inline main.py:122]: The code is mostly self-explanatory. Only need to mention what is enabled or disabled (and that should be done elsewhere).

**Comment by ADTC** [comment]
Also, better to make separate commits for unrelated changes, instead of putting everything into one commit.

**Pull request closed without merging**